### PR TITLE
TPRUN-2553 SAPJCO3 multiple endpoint support.

### PR DESCRIPTION
**Pull request by andreasmattes**

### CODE REVIEW

File: talend-sapjco3-connector/src/main/java/org/talend/sap/connection/SAPJCo3Connector.java [hunk 1]
```diff
@@ -35,15 +42,45 @@ public void setConnectionPoolName(String connectionPoolName) {
...
         Method regMethod = cls.getMethod("unregisterDestination", String.class);
-        regMethod.invoke(cls, connectionPoolName);
+        for (String ep : endpointNames) { 
+            regMethod.invoke(cls, ep);
+        }
     }
 
+    private Map<String, Properties> evaluateProperties() {
+        Map<String, Properties> result = new LinkedHashMap<>();
+        for (Map.Entry<Object, Object> e : properties.entrySet()) {
```
Comment (jgreffe) [inline talend-sapjco3-connector/src/main/java/org/talend/sap/connection/SAPJCo3Connector.java:66]: Potential NPE

File: talend-sapjco3-connector/src/main/java/org/talend/sap/connection/SAPJCo3Connector.java [hunk 2]
```diff
@@ -35,15 +42,45 @@ public void setConnectionPoolName(String connectionPoolName) {
...
+            endpointNames.add(ep);
+        }
     }
 
     public void destroy() throws Exception {
-        Class cls = Class.forName("org.hibersap.execution.jco.JCoEnvironment");
+        Class<?> cls = Class.forName("org.hibersap.execution.jco.JCoEnvironment");
         Method regMethod = cls.getMethod("unregisterDestination", String.class);
-        regMethod.invoke(cls, connectionPoolName);
+        for (String ep : endpointNames) {
```
Comment (jgreffe) [inline talend-sapjco3-connector/src/main/java/org/talend/sap/connection/SAPJCo3Connector.java:59]: Potential NPE

File: talend-sapjco3-connector/src/main/java/org/talend/sap/connection/SAPJCo3Connector.java [hunk 3]
```diff
@@ -35,15 +42,45 @@ public void setConnectionPoolName(String connectionPoolName) {
...
-        Class cls = Class.forName("org.hibersap.execution.jco.JCoEnvironment");
+        endpointNames = new ArrayList<>();
+        Map<String, Properties> endpoints = evaluateProperties();
+        Class<?> cls = Class.forName("org.hibersap.execution.jco.JCoEnvironment");
         Method regMethod = cls.getMethod("registerDestination", String.class, Properties.class);
-        regMethod.invoke(cls, connectionPoolName, properties);
+        for (Map.Entry<String, Properties> e : endpoints.entrySet()) {
+            String ep = e.getKey();
+            regMethod.invoke(cls, ep, e.getValue());
+            endpointNames.add(ep);
```
Comment (jgreffe) [inline talend-sapjco3-connector/src/main/java/org/talend/sap/connection/SAPJCo3Connector.java:52]: Potential NPE

**Review by jgreffe — CHANGES_REQUESTED** [review]

File: talend-sapjco3-connector/src/main/java/org/talend/sap/connection/SAPJCo3Connector.java (hunk 1 above)
Comment (andreasmattes) [reply talend-sapjco3-connector/src/main/java/org/talend/sap/connection/SAPJCo3Connector.java:75]: caught upfront.

File: talend-sapjco3-connector/src/main/java/org/talend/sap/connection/SAPJCo3Connector.java (hunk 2 above)
Comment (andreasmattes) [reply talend-sapjco3-connector/src/main/java/org/talend/sap/connection/SAPJCo3Connector.java:65]: caught upfront.

File: talend-sapjco3-connector/src/main/java/org/talend/sap/connection/SAPJCo3Connector.java (hunk 3 above)
Comment (andreasmattes) [reply talend-sapjco3-connector/src/main/java/org/talend/sap/connection/SAPJCo3Connector.java:55]: caught upfront

**Pull request merged**